backend/src/ml_space_lambda/initial_app_config/lambda_function.py

```diff
@@ -44,5 +46,24 @@ def lambda_handler(event, context):
     ]
 
     app_configuration_dao.update(config)
+    update_dynamic_roles_with_notebook_policies()
 
     generate_html_response(200, "Successfully updated app config")
+
+
+def update_dynamic_roles_with_notebook_policies():
+    env_vars = get_environment_variables()
+
+    if env_vars["MANAGE_IAM_ROLES"] == "false":
+        return
+
+    iam_manager = IAMManager(iam)
+    role_names = iam_manager.find_dynamic_user_roles()
+    policy_arns = [
+        env_vars["JOB_INSTANCE_CONSTRAINT_POLICY_ARN"],
+        env_vars["ENDPOINT_CONFIG_INSTANCE_CONSTRAINT_POLICY_ARN"],
+    ]
+    iam_manager.attach_policies_to_dynamic_user_roles(policy_arns, role_names)
```

Comment:
I would do one of the following:
- Update the function name to `attach_policies_to_roles` OR
- Get the list of dynamic role names within the function

I don't currently anticipate any cases we would want to attach to any other roles, so I think option 2 of combining these makes more sense.

Reply:
I'm going to take your first suggestion and change it to `attach_policies_to_roles` because if you look below I'm using the `role_names` again. This is a relatively costly thing to query for so I dont' want to do that multiple times if not necessary. I thought giving the option to supply them so they arent' fetched dynamically would be good but I agree it is slightly weird.